Failed to parse JSON

Hi,

just tested the cache machanism ad got problems:
```
requesting URI: https://services.nvd.nist.gov/rest/json/cves/2.0?resultsPerPage=2000&startIndex=214000
requesting URI: https://services.nvd.nist.gov/rest/json/cves/2.0?resultsPerPage=2000&startIndex=216000
requesting URI: https://services.nvd.nist.gov/rest/json/cves/2.0?resultsPerPage=2000&startIndex=218000
requesting URI: https://services.nvd.nist.gov/rest/json/cves/2.0?resultsPerPage=2000&startIndex=220000
Ticket returned At: 12:27:07; count: 2
Ticket taken At: 12:27:07; count: 3
Requested At: 12:27:07; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=4000
Content-Type Received: application/json
Ticket returned At: 12:27:08; count: 3
Ticket taken At: 12:27:08; count: 4
Requested At: 12:27:08; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=6000
Content-Type Received: application/json
Ticket returned At: 12:27:09; count: 4
Ticket taken At: 12:27:09; count: 5
Requested At: 12:27:09; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=8000
Content-Type Received: application/json
Ticket returned At: 12:27:12; count: 5
Ticket taken At: 12:27:12; count: 6
Requested At: 12:27:12; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=10000
Content-Type Received: application/json
Ticket returned At: 12:27:13; count: 6
Ticket taken At: 12:27:13; count: 7
Requested At: 12:27:13; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=12000
Content-Type Received: application/json
Ticket returned At: 12:27:14; count: 7
Ticket taken At: 12:27:14; count: 8
Requested At: 12:27:14; URI: /rest/json/cves/2.0?resultsPerPage=2000&startIndex=14000
Ticket returned At: 12:27:14; count: 8
waiting for 594
request failed
org.apache.hc.core5.http.RequestNotExecutedException: Connection is closed
        at org.apache.hc.core5.http.nio.command.CommandSupport.cancelCommands(CommandSupport.java:68)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onDisconnect(AbstractHttp1StreamDuplexer.java:410)
        at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.disconnected(AbstractHttp1IOEventHandler.java:95)
        at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.disconnected(ClientHttp1IOEventHandler.java:41)
        at org.apache.hc.core5.reactor.ssl.SSLIOSession$1.disconnected(SSLIOSession.java:247)
        at org.apache.hc.core5.reactor.InternalDataChannel.disconnected(InternalDataChannel.java:204)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.processClosedSessions(SingleCoreIOReactor.java:231)
        at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:133)
        at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:86)
        at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
        at java.base/java.lang.Thread.run(Thread.java:829)

ERROR
io.github.jeremylong.openvulnerability.client.nvd.NvdApiException: Failed to parse JSON starting with: "��`I�%&/m�J�J��t�`$��@�������iG#)�*��eVe]f@������{����{����;�N'���?\fdl��J���!�"
        at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:311)
        at io.github.jeremylong.vulnz.cli.commands.CveCommand.processRequest(CveCommand.java:246)
        at io.github.jeremylong.vulnz.cli.commands.CveCommand.timedCall(CveCommand.java:204)
        at io.github.jeremylong.vulnz.cli.commands.TimedCommand.call(TimedCommand.java:36)
        at io.github.jeremylong.vulnz.cli.commands.TimedCommand.call(TimedCommand.java:25)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at io.github.jeremylong.vulnz.cli.Application.run(Application.java:73)
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:768)
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:314)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1303)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1292)
        at io.github.jeremylong.vulnz.cli.Application.main(Application.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
Caused by: com.fasterxml.jackson.core.JsonParseException: Illegal character ((CTRL-CHAR, code 31)): only regular white space (\r, \n, \t) is allowed between tokens
 at [Source: (StringReader); line: 1, column: 2]
        at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:735)
        at com.fasterxml.jackson.core.base.ParserMinimalBase._throwInvalidSpace(ParserMinimalBase.java:713)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._skipWSOrEnd(ReaderBasedJsonParser.java:2461)
        at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:698)
        at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4761)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4667)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3597)
        at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient.next(NvdCveClient.java:309)
        ... 27 common frames omitted

```

Looks a bit like #42 - but I think that should have been fixed with a retry.

Andreas

--- a/open-vulnerability-clients/src/main/java/io/github/jeremylong/openvulnerability/client/nvd/NvdCveClient.java
+++ b/open-vulnerability-clients/src/main/java/io/github/jeremylong/openvulnerability/client/nvd/NvdCveClient.java
@@ -297,7 +297,6 @@ public class NvdCveClient implements PagedDataSource<DefCveItem> {
             SimpleHttpResponse response = call.getResponse();
             if (response.getCode() == 200) {
                 LOG.debug("Content-Type Received: {}", response.getContentType());
-                this.indexesToRetrieve.remove(call.getStartIndex());
                 json = response.getBodyText();
                 // resolve issue #20
                 if (json == null && response.getBody().isBytes()) {
@@ -307,6 +306,7 @@ public class NvdCveClient implements PagedDataSource<DefCveItem> {
                 CveApiJson20 current;
                 try {
                     current = objectMapper.readValue(json, CveApiJson20.class);
+                    this.indexesToRetrieve.remove(call.getStartIndex());
                 } catch (JsonProcessingException e) {
                     throw new NvdApiException("Failed to parse JSON starting with: \"" + json.substring(0, 100) + "\"",
                             e);

